File: data_structures_js/LinkedList.js
class Node {
	constructor(value=null, next=null) {
		this.value = value;
		this.next = next;
	}
}
class LinkedList {
	constructor () {
		this.head = new Node();
	}
	insert(value) {
		this.head.next = new Node(value, this.head.next);
	}
	delete() {
		if (this.head.next != null) {
			this.head.next = this.head.next.next;
		}
	}
	printList() {
		let temp = this.head.next;
		while (temp != null) {
			console.log(temp.value);
			temp = temp.next;
		}
	}
}

if (require.main == module) {
	ll = new LinkedList();
	ll.insert(34);
	ll.insert(4);
	ll.insert(14);
	ll.delete();

	ll.printList();
}

File: data_structures_java/LinkedList.java
class Node<T> {
	T value;
	Node next;
	Node () {
		next = null;
	}
	Node (T value) {
		this(value, null);
	}
	Node (T value, Node next) {
		this.value = value;
		this.next = next;
	}
}

public class LinkedList<T> {
	Node<T> head;
	LinkedList () {
		head = new Node<T>();
	}
	public void insert(T value) {
		head.next = new Node<T>(value, head.next);
	}
	public void delete() {
		if (head.next != null) {
			head.next = head.next.next;
		}
	}
	public void printList() {
		Node<T> temp = head.next;
		while (temp != null) {
			System.out.println(temp.value);
			temp = temp.next;
		}
	}

	public static void main(String[] args) {
		LinkedList<Integer> ll = new LinkedList<Integer>();
		ll.insert(44);
		ll.insert(34);
		ll.insert(3);
		ll.delete();
		ll.insert(4);
		ll.insert(415);
		ll.printList();
	}
}